tests/unit/lms/services/course_test.py

```diff
@@ -91,15 +90,13 @@ def test_any_with_setting(
         assert svc.any_with_setting("group", "key", value) is expected
 
     @pytest.fixture
-    def add_courses_with_settings(self, db_session, pyramid_request):
+    def add_courses_with_settings(self, pyramid_request):
         def add_courses_with_settings(settings_set, consumer_key=None):
             if consumer_key is None:
                 consumer_key = pyramid_request.lti_user.oauth_consumer_key
 
             for settings in settings_set:
-                db_session.add(
-                    factories.Course(consumer_key=consumer_key, settings=settings)
-                )
+                factories.Course(consumer_key=consumer_key, settings=settings)
```

Comment:
Is this where we are making use of the new relationship? It's not obvious to me

Reply:
This would crash without the new relationship, yeah. Because SQLAlchemy spits out the SQL statements in the wrong order:

```
E           sqlalchemy.exc.IntegrityError: (raised as a result of Query-invoked autoflush; consider using a session.no_autoflush block if this flush is occurring prematurely)
E           (psycopg2.errors.ForeignKeyViolation) insert or update on table "course" violates foreign key constraint "fk__course__consumer_key__application_instances"
E           DETAIL:  Key (consumer_key)=(Hypothesis53fcdbb7055fb63c370fe829181b2e92) is not present in table "application_instances".
E           
E           [SQL: INSERT INTO course (consumer_key, authority_provided_id, settings) VALUES (%(consumer_key)s, %(authority_provided_id)s, %(settings)s)]
E           [parameters: ({'consumer_key': 'Hypothesis53fcdbb7055fb63c370fe829181b2e92', 'authority_provided_id': 'c146fc19badf85cb3a3a9a628b80e623eff9012b', 'settings': '{"group": {"key": "match"}}'}, {'consumer_key': 'other_consumer_key', 'authority_provided_id': '0a8decc90049cc68efc3310e8b10127dcabfd79d', 'settings': '{"group": {"key": "match"}}'})]
E           (Background on this error at: http://sqlalche.me/e/13/gkpj)
```